packages/contracts-bedrock/test/DelayedVetoable.t.sol

```diff
@@ -0,0 +1,181 @@
+        delayedVetoable.initiator();
+        vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, initiator, address(this)));
+        delayedVetoable.vetoer();
+        vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, initiator, address(this)));
+        delayedVetoable.target();
+        vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, initiator, address(this)));
+        delayedVetoable.delay();
+    }
+}
+
+contract DelayedVetoable_HandleCall_Test is DelayedVetoable_Init {
+    /// @dev A call can be initiated by the initiator.
+    function testFuzz_handleCall_initiation_succeeds(bytes memory data) external {
+        assumeNonzeroData(data);
+
+        vm.expectEmit(true, false, false, true, address(delayedVetoable));
+        emit Initiated(keccak256(data), data);
+
+        vm.prank(initiator);
+        (bool success,) = address(delayedVetoable).call(data);
+        assertTrue(success);
+    }
+
+    /// @dev The delay is inititially set to zero and the call is immediately forwarded.
+    function testFuzz_handleCall_initialForwardingImmediately_succeeds(
+        bytes memory inData,
+        bytes memory outData
+    )
+        external
+    {
+        assumeNonzeroData(inData);
+
+        // Reset the delay to zero
+        vm.store(address(delayedVetoable), bytes32(uint256(0)), bytes32(uint256(0)));
+
+        vm.mockCall(target, inData, outData);
+        vm.expectEmit(true, false, false, true, address(delayedVetoable));
+        vm.expectCall({ callee: target, data: inData });
+        emit Forwarded(keccak256(inData), inData);
+        vm.prank(initiator);
+        (bool success, bytes memory returnData) = address(delayedVetoable).call(inData);
+        assertTrue(success);
+        assertEq(returnData, outData);
+    }
+
+    /// @dev The delay can be activated by the vetoer or initiator, and are not forwarded until the delay has passed
+    ///      once activated.
+    function testFuzz_handleCall_forwardingWithDelay_succeeds(bytes memory data) external {
+        assumeNonzeroData(data);
+
+        vm.prank(initiator);
+        (bool success,) = address(delayedVetoable).call(data);
+
+        vm.warp(block.timestamp + operatingDelay);
+        vm.expectEmit(true, false, false, true, address(delayedVetoable));
+        emit Forwarded(keccak256(data), data);
+
+        vm.expectCall({ callee: target, data: data });
+        (success,) = address(delayedVetoable).call(data);
+        assertTrue(success);
+    }
+}
+
+contract DelayedVetoable_HandleCall_TestFail is DelayedVetoable_Init {
+    /// @dev The delay is inititially set to zero and the call is immediately forwarded.
+    function test_handleCall_unauthorizedInitiation_reverts() external {
+        vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, initiator, address(this)));
+        (bool success,) = address(delayedVetoable).call(NON_ZERO_DATA);
```

Comment:
I am confused here, expect revert but success is true?

Reply:
It super weird see the "gotcha" [here](https://book.getfoundry.sh/cheatcodes/expect-revert?highlight=expectrevert#description). 

I'm going to delete the `assertTrue` statement, it's just confusing and redundant with `expectRevert`.